runtime/lua/vim/treesitter/query.lua

```diff
@@ -179,11 +179,19 @@ function M.parse_query(lang, query)
   end
 end
 
---- Gets the text corresponding to a given node
+--- Gets the text corresponding to a given node or a list of nodes
 ---
----@param node the node
+---@param node the node or list of nodes
 ---@param source The buffer or string from which the node is extracted
 function M.get_node_text(node, source)
+  -- If node is a list of nodes then return their concatenated text
+  if node['start'] == nil and type(node) == 'table' then
```

Comment:
Shouldn't these conditions be in reverse order? You can't check an index is present without knowing the variable is a table first.

```suggestion
  if type(node) == 'table' and node['start'] == nil then
```

Reply:
Even better:
```suggestion
  if  type(node) == 'table' and not node.start then
```